### DISCUSSION TITLE
How to properly include script intended to run on client with SSG?

### DISCUSSION THREAD

**Discussion by Gusterson** [vercel/next.js#13502] [Help] [answered]
Attempting to build a fully static app with some non-react vanilla JS on top to manipulate the DOM for a better a UX.

How do I best include the necessary JS that should run on load time?

- I'm assuming I put a `<script src="my-script.js"></script>` in the `next/head` component?
- Can that script live anywhere? Or does it need to be in the `/public` folder?
- Will it be magically optimized / minified etc by Next during SSG?

So many questions... but I think that's a start. Thanks!

**Comment by Gusterson** [comment, marked as answer]
OK disregard this ... Turns out nothing script-related was working because I was testing Next.js in a subdirectory: `domain/nextjs` which Next,js doesn't have support for yet. :P

**Comment by kr8n3r** [comment]
@Gusterson did you manage to get the client-side script running with SSG?

**Reply by Gusterson** [reply]
It ended up being a non-issue for me, but I'd still be curious about the original question... (I deleted the tangent question to keep this thread focused).

**Reply by kr8n3r** [reply]
it seems you cannot run it properly.
- you cannot put it `_document` as that is rendered by the server
- you can put it elsewhere, but because pages are rendered server-side with client-side transition, script won't reliable run on all pages - only on those that do a proper http request
- if you use `next/head` or` react-helmet` in any way to modify `html` or `body` with classNames or attributes, external script won't run at all

**Comment by gentlee** [comment]
Better to use `next/script` instead of `next/head`.
